scripts/src/main/resources/scripts/functions

```diff
@@ -1824,53 +1830,142 @@ function doGetSoftwareVersion() {
   fi
 }
 
+# $1: software
+function doGetSoftwareEdition() {
+  local software="${1}"
+  local edition
+  echo "${software}" | grep '-' >/dev/null
+  ERG="$?"
+  if [ "${ERG}" = "0" ]
+  then
+    echo "${software}"
+  else
+    edition="$(echo "${software}" | awk '{print toupper($0)}')_EDITION"
+    if [ "$(printf '%s\n' "${!edition}")" == "" ]
+    then
+      echo "${software}"
+    else
+      echo "${!edition}"
+    fi
+  fi
+}
+
+# $1: software
+# $2: optional prefix
+function doGetLatestSoftwareVersion() {
+  local software="${1}"
+  local prefix="${2}"
+  local edition
+  local version
+  doUpdateUrls
+  edition=$(doGetSoftwareEdition "${software}")
+  if [ -z "${prefix}" ]
+  then
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -printf "%f\n" | sort -rV | head -1)"
+  else
+    prefix="${prefix:0:${#prefix}-1}"
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -printf "%f\n" | grep "^${prefix}" | sort -rV | head -1)"
+  fi
+  echo "${version}"
+} 
+
 # $1: software
 # $2: selected version
 # $3: commandlet name
 function doSetSoftwareVersion() {
   local software="${1}"
   local version="${2}"
   local commandlet="${3}"
-  if [ -n "${version}" ]
+  local edition
+  if [ -d "${DEVON_IDE_HOME}/urls" ]
   then
-    if [ "${version}" = "latest" ]
+    if [ -n "${version}" ]
     then
-      doUpdateMirrors
-      version="$(head -n 1 "${DEVON_IDE_HOME}/mirrors/${software}/available-versions")"
+      if [ "${version}" = "latest" ]
+      then
+        doUpdateUrls
+        edition=$(doGetSoftwareEdition "${software}")
+        version=doGetLatestSoftwareVersion "${software}"
```

Comment:
how can this work? A bash function can only return an exit code.
Did you forget `$(...)`?

Reply:
For the record: you determine the edition here via `doGetSoftwareEditon` and the very next call `doGetLatestSoftwareVersion` does not get the edition as argument but again has to compute it again via the same `doGetSoftwareEditon` function. With every day I am getting more keen on our migration to Java...